pkg/apk/impl/repo.go

```diff
@@ -310,10 +278,37 @@ func (p *PkgResolver) GetPackagesWithDependencies(packages []string) (toInstall
 	return toInstall, conflicts, nil
 }
 
-// GetPackageDependencies get all of the dependencies for a single package based on the
-// indexes.
-func (p *PkgResolver) GetPackageDependencies(pkg string) (dependencies []*repository.RepositoryPackage, conflicts []string, err error) {
+// GetPackageDependencies get all of the dependencies for a single package as well as looking
+// up the package itself and resolving its version, based on the indexes.
+func (p *PkgResolver) GetPackageWithDependencies(pkgName string) (pkg *repository.RepositoryPackage, dependencies []*repository.RepositoryPackage, conflicts []string, err error) {
 	parents := make(map[string]bool)
+
+	name, version, compare := resolvePackageNameVersion(pkgName)
+	pkgsWithVersions, ok := p.nameMap[name]
+
+	if !ok {
+		trueName, ok := p.providesMap[name]
+		if !ok {
+			return nil, nil, nil, fmt.Errorf("could not find package, alias or a package that provides %s in indexes", pkgName)
+		}
+		pkgsWithVersions, ok = p.nameMap[trueName]
+		if !ok {
+			return nil, nil, nil, fmt.Errorf("looking for %s, found provided by %s, but could not find %s in indexes", pkgName, trueName, trueName)
+		}
+	}
+
+	// pkgsWithVersions contains a map of all versions of the package
+	// get the one that most matches what was requested
+	versions := make([]string, 0, 10)
```

Comment:
significance of size 10? imagine this could grow bigger?

Reply:
Nothing really, just a starting size from which it could grow. lint insists you give it something, 10 seemed as reasonable as any other number.